### DISCUSSION TITLE
How to use locally modified @atproto/api with social-app?

### DISCUSSION THREAD

**Discussion by Nazgul07** [bluesky-social/atproto#2669] [How to do it?] [answered]
I have made updates to @atproto/api (rich-text) in my fork, and I have used "yarn link" to link the @atproto/api package reference in social-app to point to my local package/copy. 

I am running the atproto dev server.

My IDE picks up the updated api signature's, but running social-app indicates to me that it is not picking up the modified api at all.

Is there another approach I should be using other than "yarn link"?

When running social-app, it doesn't seem to be using the local dev server as well...

**Comment by bnewbold** [comment, marked as answer]
Very reasonable question! I don't work with the JS ecosystem as much, and don't have a reliable answer for you. We do have some docs for linking when doing Ozone development against a patched version of the api package:

https://github.com/bluesky-social/ozone/blob/main/HACKING.md

We are planning to change how the api package works to make it more app-agnostic, and improve docs on building and working with independently-designed Lexicon schemas in the near future. Maybe that will help!

**Reply by Nazgul07** [reply]
I'm marking this as the answer because it seems to be correct. I'm realizing now that the social-app doesn't seem to actually hook up to the dev-env just because I am running it, it's always pulling from the live public urls, and I have yet to figure out how to make it use the local app view server from the dev-env.

**Comment by itaru2622** [comment]
@bnewbold what should be done when lexicon modified?

social-app has golang code in social-app/bskyweb/cmd/bsyweb/server.go
which uses indigo's bsky api by refering github.com/bluesky-social/indigo/api/bsky
so, social-app may need additional care than ozone.

in case of ozone,  ozone doesn't have golang code so HACKING.md works.
